src/messaging/BUILD.gn

```diff
@@ -77,8 +77,4 @@ static_library("messaging") {
     "${chip_root}/src/transport",
     "${chip_root}/src/transport/raw",
   ]
-
-  if (chip_enable_icd_server) {
-    public_deps += [ "${chip_root}/src/app/icd:manager" ]
-  }
```

Comment:
I think this might brake the MRP increase based on polling intervals looking at the unit tests.

Reply:
fixed with defines

Reply:
This remove was not ok: ICDManager is included by the CPP files, so you cannot just remove the dependency.

See #30527